Debonair cilantro: Deleted Projects component

modernserf opened the pull request:
## Links
* Remix link https://debonair-cilantro.glitch.me
* Manuscript link https://glitch.manuscript.com/f/cases/3328433/

## GIF/Screenshots:
hidden
![Screen Shot 2019-04-26 at 4 03 53 PM](https://user-images.githubusercontent.com/938722/56834817-7c0e0500-6840-11e9-8470-2fa79601f190.png)
active (top)
![Screen Shot 2019-04-26 at 4 04 47 PM](https://user-images.githubusercontent.com/938722/56834818-7c0e0500-6840-11e9-985f-5735988407e2.png)
active (bottom)
![Screen Shot 2019-04-26 at 4 04 54 PM](https://user-images.githubusercontent.com/938722/56834819-7c0e0500-6840-11e9-8c9e-48cbd84c2cfb.png)

## Changes:
* componentize Deleted Projects
* use grid auto-fill instead of media queries
* make TransparentButton component for accessible, clickable container
* move Deleted Projects header to user page

## How To Test:
* visit https://debonair-cilantro.glitch.me
* log in
* view the deleted projects on your profile page
* un-delete some projects

taravancil commented on src/components/deleted-projects/index.js:
```diff
@@ -0,0 +1,107 @@
+    const node = ref.current;
+    if (!node) return;
+    node.addEventListener('animationend', () => undeleteTracked(id), { once: true });
+    node.classList.add('slide-up');
```
Looks like this animation isn't working. When I click undelete, there's a short delay, then the project just disappears from the list

modernserf commented:
@taravancil updated:
- fixed bug with only deleting one at a time
- added `isTitle` to Emoji component for consistency
- created AnimationContainer to correctly handle animations with components
